runtime error: index out of range when site repication configuration

## NOTE

Created 3 Minio single node helm installs with **MinIO VERSION  2022-08-13T21:54:44Z**

when creating site replication from the only minio with a bucket created I got this error in minio logs(the one I trigered add replication from console UI

```
panic: "PUT /minio/admin/v3/site-replication/add": runtime error: index out of range [-1]                                                          │
│ goroutine 272884 [running]:                                                                                                                        │
│ runtime/debug.Stack()                                                                                                                              │
│     runtime/debug/stack.go:24 +0x65                                                                                                                │
│ github.com/minio/minio/cmd.setCriticalErrorHandler.func1.1()                                                                                       │
│     github.com/minio/minio/cmd/generic-handlers.go:578 +0x8e                                                                                       │
│ panic({0x2621420, 0xc0137327f8})                                                                                                                   │
│     runtime/panic.go:838 +0x207                                                                                                                    │
│ github.com/minio/minio/cmd.(*SiteReplicationSys).AddPeerClusters(0x4d9cde0?, {0x4d9cde0, 0xc002245c50}, {0xc0005f7a00?, 0x2281840?, 0xc000e767f8?} │
│     github.com/minio/minio/cmd/site-replication.go:414 +0x1850                                                                                     │
│ github.com/minio/minio/cmd.adminAPIHandlers.SiteReplicationAdd({}, {0x4d99080, 0xc002f16be0}, 0xc0005f7900)                                        │
│     github.com/minio/minio/cmd/admin-handlers-site-replication.go:55 +0x2fc                                                                        ││ net/http.HandlerFunc.ServeHTTP(...)                                                                                                                ││     net/http/server.go:2084                                                                                                                        ││ github.com/minio/minio/cmd.httpTrace.func1({0x4d99080, 0xc002f16be0}, 0xc0005f7900)                                                                ││     github.com/minio/minio/cmd/http-tracer.go:238 +0x1af                                                                                           ││ net/http.HandlerFunc.ServeHTTP(0x2313780?, {0x4d99080?, 0xc002f16be0?}, 0x4?)                                                                      │
│     net/http/server.go:2084 +0x2f                                                                                                                  │
│ github.com/klauspost/compress/gzhttp.NewWrapper.func1.1({0x4d99080, 0xc002f16be0}, 0xc00102ae00?)                                                  ││     github.com/klauspost/compress@v1.15.9/gzhttp/compress.go:391 +0x347                                                                            ││ net/http.HandlerFunc.ServeHTTP(0x7fe59d4d4200?, {0x4d99080?, 0xc002f16be0?}, 0x417aeb?)                                                            ││     net/http/server.go:2084 +0x2f                                                                                                                  ││ github.com/minio/minio/cmd.setBucketForwardingHandler.func1({0x4d99080, 0xc002f16be0}, 0xc0005f7900)                                               ││     github.com/minio/minio/cmd/generic-handlers.go:486 +0x22d                                                                                      ││ net/http.HandlerFunc.ServeHTTP(0x4d99020?, {0x4d99080?, 0xc002f16be0?}, 0x10?)                                                                     ││     net/http/server.go:2084 +0x2f                                                                                                                  ││ github.com/minio/minio/cmd.addCustomHeaders.func1({0x4d99020, 0xc000e766c0}, 0xc01576edc0?)                                                        ││     github.com/minio/minio/cmd/generic-handlers.go:561 +0x3c2                                                                                      ││ net/http.HandlerFunc.ServeHTTP(0x26b7340?, {0x4d99020?, 0xc000e766c0?}, 0x39?)                                                                     ││     net/http/server.go:2084 +0x2f
```

Site replication config
![image](https://user-images.githubusercontent.com/2348131/186232034-19a7ea24-62d5-4887-8f66-c7425fe51cda.png)

--- a/cmd/site-replication.go
+++ b/cmd/site-replication.go
@@ -360,6 +360,9 @@ func (c *SiteReplicationSys) AddPeerClusters(ctx context.Context, psites []madmi
 			nonLocalPeerWithBuckets = v.Name
 		}
 	}
+	if selfIdx == -1 {
+		return madmin.ReplicateAddStatus{}, errSRBackendIssue(fmt.Errorf("global deployment ID %s mismatch, expected one of %s", globalDeploymentID, deploymentIDsSet))
+	}
 	if !currDeploymentIDsSet.IsEmpty() {
 		// If current cluster is already SR enabled and no new site being added ,fail.
 		if currDeploymentIDsSet.Equals(deploymentIDsSet) {

